### PR TITLE
unit_test for class Network

### PR DESCRIPTION
**Pull request by gaetanoantonicchio**
Created unit tests for class Network

### CODE REVIEW

File: src/test_network.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        # make sure Exception is raised when there is a mismatch between targets and outputs
+        self.assertRaises(Exception, net.fit, inp=(1, 1, 1), target=(1, 0, 1))
+
+    def test_ValueError(self):
+        # make sure a ValueError is raised when input_dim or units_per_layer are not positive
+        self.assertRaises(ValueError, Network, input_dim=-2, units_per_layer=[6, 2], acts=['relu', 'relu'])
+        self.assertRaises(ValueError, Network, input_dim=2, units_per_layer=[-4, 0], acts=['relu', 'relu'])
+
+        #TODO: test ValueError if units in one layer have different act functions
+        # This assignment was done with a private method I cannot access for testing
```
Comment (AlexPasqua) [inline src/test_network.py:21]: Which private method?

File: src/test_network.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    net = Network(input_dim=3, units_per_layer=[6, 2], acts=['relu', 'relu'])
+    net.compile('sgd', 'squared', lrn_rate=0.01)
+
+    def test_ExceptionForward(self, net=net):
+        # make sure Exception is raised when there is a mismatch between input_dim and inp
+        self.assertRaises(Exception, net.forward, inp=(1, 1))
+        # make sure Exception is raised when there is a mismatch between targets and outputs
+        self.assertRaises(Exception, net.fit, inp=(1, 1, 1), target=(1, 0, 1))
+
+    def test_ValueError(self):
```
Comment (AlexPasqua) [inline src/test_network.py:15]: Chiamalo `test_constructor` vah

File: src/test_network.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import unittest
+from src.network import Network
+
+
+class TestNetwork(unittest.TestCase):
+    net = Network(input_dim=3, units_per_layer=[6, 2], acts=['relu', 'relu'])
+    net.compile('sgd', 'squared', lrn_rate=0.01)
+
+    def test_ExceptionForward(self, net=net):
```
Comment (AlexPasqua) [inline src/test_network.py:9]: Chiamalo semplicemente `test_forward`, poi ci mettiamo anche i test che possiamo fare una volta create le initializations

**Review by AlexPasqua — CHANGES_REQUESTED** [review]

File: src/test_network.py (hunk 1 above)
Comment (gaetanoantonicchio) [reply src/test_network.py:21]: def __init__(self, units):
        """
        Constructor
        :param units: list on layer's units ('Unit' objects)
        """
        units_acts = [u.act.name for u in units]
        for act in units_acts[1:]:
            if act != units_acts[0]:
                raise ValueError("All units in a layer must have the same activation function")
        self.__units = units
        self.__act = self.__units[0].act
        self.__outputs = []

How can I access the parameters of the constructor? Any suggestions?

File: src/test_network.py (hunk 2 above)
Comment (gaetanoantonicchio) [reply src/test_network.py:15]: done!

File: src/test_network.py (hunk 3 above)
Comment (gaetanoantonicchio) [reply src/test_network.py:9]: done!

File: src/test_network.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        self.assertRaises(Exception, net.forward, inp=(1, 1))
+        # make sure Exception is raised when there is a mismatch between targets and outputs
+        self.assertRaises(Exception, net.fit, inp=(1, 1, 1), target=(1, 0, 1))
+
+    def test_constructor(self):
+        # make sure a ValueError is raised when input_dim or units_per_layer are not positive
+        self.assertRaises(ValueError, Network, input_dim=-2, units_per_layer=[6, 2], acts=['relu', 'relu'])
+        self.assertRaises(ValueError, Network, input_dim=2, units_per_layer=[-4, 0], acts=['relu', 'relu'])
+
+        #TODO: test ValueError if units in one layer have different act functions
```
Comment (AlexPasqua) [inline src/test_network.py:20]: Beh aggiungi lo stesso un `assertRaises` con activation functions diverse

**Review by AlexPasqua — CHANGES_REQUESTED** [review]

**Review by AlexPasqua — APPROVED** [review]

**Pull request merged**